src/analyses/reaching_definitions.cpp

```diff
@@ -26,9 +26,34 @@ Date: February 2013
 #include "is_threaded.h"
 #include "dirty.h"
 
+/// This ensures that all domains are constructed with the appropriate pointer
+/// back to the analysis engine itself.  Using a factory is a tad verbose
+/// but it works well with the ait infrastructure.
+class rd_range_domain_factoryt : public ai_domain_factoryt<rd_range_domaint>
+{
+public:
+  rd_range_domain_factoryt(
+    sparse_bitvector_analysist<reaching_definitiont> *_bv_container)
```

Comment:
If this is never null, should it just be a reference?

Reply:
I converted at least one case of these when I was doing this, the others I didn't convert for reasons.  IIRC these were wanting to keep the patches as small and focused as possible and it's not quite that simple; there are some cases where it isn't that straight forward.

If it bothers you I can write a separate PR for it?